lib/filewatch/observing_tail.rb

```diff
@@ -1,34 +1,22 @@
-require 'filewatch/tail_base'
+require 'filewatch/boot_setup' unless defined?(FileWatch)
 
 module FileWatch
   class ObservingTail
     include TailBase
     public
 
-    class NullListener
-      def initialize(path) @path = path; end
-      def accept(line) end
-      def deleted() end
-      def created() end
-      def error() end
-      def eof() end
-      def timed_out() end
-    end
-
-    class NullObserver
-      def listener_for(path) NullListener.new(path); end
-    end
-
     def subscribe(observer = NullObserver.new)
       @watch.subscribe(@opts[:stat_interval],
                        @opts[:discover_interval]) do |event, watched_file|
         path = watched_file.path
         file_is_open = watched_file.file_open?
         listener = observer.listener_for(path)
+        @logger.debug? && @logger.debug("subscribe block - #{event} for #{path}")
         case event
         when :unignore
-          listener.created
-          _add_to_sincedb(watched_file, event) unless @sincedb.member?(watched_file.inode)
+          if !@sincedb.member?(watched_file.storage_key) && !file_is_open && _open_file(watched_file, event)
```

Comment:
`@sincedb.member?(watched_file.storage_key)` could be just `member?(watched_file.storage_key)`